src/Persistence/PaYaml/Serialization/PaYamlSerializer.cs

```diff
@@ -115,6 +115,10 @@ private static void WriteTextWriter<TValue>(TextWriter writer, in TValue? value,
 
             return value;
         }
+        catch (YamlException ex) when (ex.InnerException is ArgumentException && ex.InnerException.Message.Contains("An item with the same key has already been added"))
```

Comment:
note: I purposely left this special handling out as it's assuming things which could produce incorrect error messages.
But this change is simple enough to be changed later iif needed.